### PR TITLE
SMT creation operation reimplemented with collateral changes.

### PR DESCRIPTION
**Pull request by Kiwonik**
Covers both #1554 and #1622 and is needed for further development, hence request to merge it ASAP.

### CODE REVIEW

File: libraries/protocol/include/steem/protocol/smt_operations.hpp [hunk 1]
```diff
@@ -8,23 +8,25 @@
...
 /**
- * This operation elevates given steem account to SMT control account on conditions that:
- * a) the account is not SMT control account already and
- * b) successful fee transfer occurs.
+ * This operation introduces new SMT into blockchain as identified by
+ * Numerical Asset Identifier (NAI). Also the SMT precision (decimal points)
+ * is explicitly provided.
  */
-struct smt_elevate_account_operation : public base_operation
+struct smt_create_operation : public base_operation
```
Comment (mvandeberg) [inline libraries/protocol/include/steem/protocol/smt_operations.hpp]: Please include the precision as well for clarity and redundancy.

Comment (Kiwonik) [reply libraries/protocol/include/steem/protocol/smt_operations.hpp]: Ain't decimal places the equivalent of precision? If you want `decimal_places` field to be removed and `precision` field to be added, what type should the `precision` be, other than `uint8_t` of `decimal_places`?

**Comment by Kiwonik** [comment]
Ignore my last question - my mistake. All requested changes are introduced now. Ready to merge again.

File: libraries/chain/include/steem/chain/smt_objects/smt_token_object.hpp [hunk 2]
```diff
@@ -63,14 +69,17 @@ class smt_token_object : public object< smt_token_object_type, smt_token_object
...
+struct by_nai;
 struct by_control_account;
 
 typedef multi_index_container <
    smt_token_object,
    indexed_by <
       ordered_unique< tag< by_id >,
          member< smt_token_object, smt_token_id_type, &smt_token_object::id > >,
-      ordered_unique< tag< by_control_account >,
+      ordered_unique< tag< by_nai >,
```
Comment (mvandeberg) [inline libraries/chain/include/steem/chain/smt_objects/smt_token_object.hpp]: We could probably make this index by asset symbol type, but I don't think it is a blocker for this merge.

Comment (mvandeberg) [reply libraries/chain/include/steem/chain/smt_objects/smt_token_object.hpp]: Later PRs are going to modify this heavily.

File: libraries/protocol/include/steem/protocol/smt_operations.hpp [hunk 3]
```diff
@@ -166,6 +170,8 @@ struct smt_emissions_unit
 
 struct smt_setup_emissions_operation : public base_operation
 {
+   /// Contains Numerical Asset Identifier (NAI) of the SMT.
+   uint32_t            nai;
```
Comment (mvandeberg) [inline libraries/protocol/include/steem/protocol/smt_operations.hpp]: Any cross reference to the SMT should probably be by the asset num. The NAI is a string only representation.

**Pull request merged**